scripts/build/docker/run-command.sh

```diff
@@ -89,6 +89,7 @@ done
 # The `-t` and TERM setup is needed to display coloured output.
 docker run -t \
   --env "TERM=xterm-256color" \
+  --env "TARGET_ARCH=$ARGV_ARCHITECTURE" \
```

Comment:
Ahhh, I remember reading something about that on the Docker Ubuntu images we were using.
Did a bit of searching and found https://github.com/docker-32bit/ubuntu/issues/18 which says "I think that it makes no sense to run a 32bit container in a 64bit host" :-/
https://hub.docker.com/r/32bit/ubuntu/ and https://github.com/docker-32bit/ubuntu/issues/12 seem to contain more info.

I guess setting the TARGET_ARCH env-var is just as effective as setting the Docker entrypoint though. Have you checked that the 32-bit files that this generates do indeed run in a pure 32-bit VM? (not just a 32-bit container using a 64-bit kernel).

Although thinking about it a bit more, if we're supplying the $TARGET_ARCH anyway, is there any benefit to using separate containers, rather than just running everything in a single 64-bit container?  This mixing of arches and containers and builds gets a bit confusing... ;-)

...

Or is this only an issue for local-development, and things actually _would_ work as expected on Travis without this "TARGET_ARCH=$ARGV_ARCHITECTURE" since it does run the x86 build in an x86 VM? (In which case, maybe we could check for the CI variable, and only manually override the TARGET_ARCH if CI isn't set?)